### ISSUE TITLE
Can't import all operators in NodeJS

### ISSUE THREAD

**Issue by stefanholzapfel** [kofrasa/mingo#144]
Hi!

I'm on 
NodeJS version 14, 
Mingo version 3.0.2

I bundle the app with Webpack version 5 (also tried with 4, same effect) and run it via nodemon (2.0.4).

I try to import all operators like that (bundle size is no problem for me):

```
import { OperatorType, useOperators } from 'mingo/core';
import * as pipelineOperators from 'mingo/operators/pipeline';
import * as projectionOperators from 'mingo/operators/projection';
import * as queryOperators from 'mingo/operators/query';
import * as expressionOperators from 'mingo/operators/expression';
import * as accumulatorOperators from 'mingo/operators/accumulator';

useOperators(OperatorType.PIPELINE, pipelineOperators);
useOperators(OperatorType.PROJECTION, projectionOperators);
useOperators(OperatorType.QUERY, queryOperators);
useOperators(OperatorType.EXPRESSION, expressionOperators);
useOperators(OperatorType.ACCUMULATOR, accumulatorOperators);    
``` 

Some of the imports work, some (like in this example arithmetic) throw:
```
(function (exports, require, module, __filename, __dirname) { export * from './arithmetic';
                                                              ^^^^^^
SyntaxError: Unexpected token 'export'
    at new Script (vm.js:99:7)
    at createScript (vm.js:247:10)
    at Object.runInThisContext (vm.js:295:10)
    at wrapSafe (internal/modules/cjs/loader.js:1104:15)
    at Module._compile (internal/modules/cjs/loader.js:1164:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
[nodemon] app crashed - waiting for file changes before starting...

```
Other es6 imports work fine in my app. It also seem to have somehting to do with import ordering (which is weird)?

Any ideas?

Thanks!

**Comment by j** [comment]
Not sure if this is unrelated, but I couldn't get tests to start b/c this 3.x branch of mingo.  I would get a memory leak somehow.  Reverting back to the older version of mingo fixes my issue.

**Comment by stefanholzapfel** [comment]
To be more precise, I get errors with:
`useOperators(OperatorType.EXPRESSION, expressionOperators);`
and
`useOperators(OperatorType.ACCUMULATOR, accumulatorOperators);`

Other operators can be imported.

**Comment by kofrasa** [comment]
@stefanholzapfel  thanks for reporting the issues. The error seems related to module loading. Starting from `3.0.0` there is a dependency on the [esm](https://www.npmjs.com/package/esm) which allows using the ES6 import syntax, which is the target of the Typescript build.

If you are getting a syntax error, my guess is that `esm` is not installed in your environment. Assuming you got that working correctly, then something else must be causing the import of `expression` and `accumulator` operators to fail.

Can you provide the exception for those cases and how are you bundling your project?

**Comment by kofrasa** [comment]
> Not sure if this is unrelated, but I couldn't get tests to start b/c this 3.x branch of mingo. I would get a memory leak somehow. Reverting back to the older version of mingo fixes my issue.

@j please open a new issue with the details of the memory leak.

**Comment by j** [comment]
@kofrasa https://github.com/kofrasa/mingo/issues/145 might be related.

**Comment by kofrasa** [comment]
I am unable to repro the import errors. Below is my example using `esm` from my terminal.

```js
➜  ~ node -r esm
Welcome to Node.js v14.4.0.
Type ".help" for more information.
> import 'mingo/init/system'
undefined
> import * as pkg from 'mingo/package.json'
undefined
> pkg.version
'3.0.2'
> import { Query } from 'mingo/query'
undefined
> let q = new Query({a: {$gt: 5}})
undefined
> q
Query {
  __criteria: { a: { '$gt': 5 } },
  __config: { idKey: '_id' },
  __compiled: [ [Function (anonymous)] ]
}
> q.test({a:10})
true
> q.test({a:1})
false
>
> import * as expressionOperators from 'mingo/operators/expression'
undefined
> useOperators(OperatorType.EXPRESSION, expressionOperators)
undefined
>
```

**Comment by stefanholzapfel** [comment]
Hi Kofrasa!

Yes you are right, I don't have esm installed, and its a Typescript build via Webpack.

I don't know the different module builders very well, but I don't unterstand why the ES export syntax works with pipeline, projection and query operators and not with accumulators and expressions. I also use other packages with ES syntax...

The exact error for expression operators is:

```
C:\xxxx\node_modules\mingo\operators\accumulator\index.js:4
export * from './addToSet';
^^^^^^

SyntaxError: Unexpected token 'export'
    at new Script (vm.js:99:7)
    at createScript (vm.js:247:10)
    at Object.runInThisContext (vm.js:295:10)
    at wrapSafe (internal/modules/cjs/loader.js:1104:15)
    at Module._compile (internal/modules/cjs/loader.js:1164:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
[nodemon] app crashed - waiting for file changes before starting...
```
Thank you!

**Comment by kofrasa** [comment]
I am unable to repro this but the latest version [3.0.3](https://github.com/kofrasa/mingo/releases/tag/3.0.3) removes the reliance on the `esm` module which is likely the root cause. 

Closing this issue. Please reopen if the issue still persists after upgrade.

**Comment by stefanholzapfel** [comment]
Thanks, fixed with new version.
